spec/controllers/saml_post_controller_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+describe SamlPostController do
+  describe 'POST /api/saml/auth' do
+    render_views
+    include ActionView::Helpers::FormTagHelper
+
+    let(:form_action_regex) { /<form.+action=".+\/api\/saml\/authpost\d{4}.+"/ }
+    let(:saml_request) { 'abc123' }
+    let(:relay_state) { 'def456' }
+    let(:sig_alg) { 'aes256' }
+    let(:signature) { 'xyz789' }
+
+    it 'renders the appropriate form' do
+      post :auth, params: {
+        'SAMLRequest' => saml_request,
+        'RelayState' => relay_state,
+        'SigAlg' => sig_alg,
+        'Signature' => signature,
+      }
+
+      expect(response.body).to match(form_action_regex)
+      expect(response.body).to match(hidden_field_tag('SAMLRequest', saml_request))
+      expect(response.body).to match(hidden_field_tag('RelayState', relay_state))
+      expect(response.body).to match(hidden_field_tag('SigAlg', sig_alg))
+      expect(response.body).to match(hidden_field_tag('Signature', signature))
+    end
+
+    it 'does not render extra parameters' do
+      post :auth, params: { 'Foo' => 'bar' }
+
+      expect(response.body).not_to match(hidden_field_tag('Foo', 'bar'))
+    end
+
+    it 'leaves the session cookie alone' do
```

Comment:
This does not appear to be doing what I want it to do since it passes when I remove the `after_action`.

Reply:
Yeah, this one is going to be hard since a lot of this behavior is going to be mocked out in the controller spec. I think our best bet may be to test that the request env is modified the way we expect by the controller.

We could potentially add a request spec to verify that it does not send an upaya session cookie?

Reply:
ah you mean just look for the `set-cookie` header? I think that works

Reply:
This worked great, good suggestion!